### DISCUSSION TITLE
Show time relative to cycle length instead of day

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#26] [Feature Requests]
## What

NP always converts time to day hour minute seconds format, which can be annoying when researching banking and experimentation. It would make it a lot easier if the extension either replaced the ETA field with one that does time relative to cycle length, or even better added another field. ![cycle research example](https://user-images.githubusercontent.com/68225269/204903398-59087b74-caaf-444f-949c-78b7698fba6f.png)
(Ignore the handwritten 'here' XD. I couldn't figure out how to add text with the snipping tool)

## How

Time is formatted using the formatTime function in crux.js at line 1081
```javascript
Crux.formatTime = function (ms, mins, secs) {
        var seconds = ms / 1000;
        var output = "";
        var minutes = 0;
        var hours = 0;
        var days = 0;
        var total_seconds = seconds;

        if (seconds >= 86400) {
            days = seconds / 86400;
            seconds = seconds % 86400;
            output += parseInt(days, 10) + "d ";
        }

        if (seconds >= 3600) {
            hours = seconds / 3600;
            seconds = seconds % 3600;
            output += parseInt(hours, 10) + "h ";
        }

        if (seconds >= 60) {
            minutes = seconds / 60;
            seconds = seconds % 60;
            if (mins) {
                output += parseInt(minutes, 10) + "m ";
            }
        }

        if (seconds > 0) {
            if (mins && secs) {
                output += parseInt(seconds, 10) + "s";
            }
        }

        return output.trim();
```
Seems like you could just create a similar function that uses cycle length instead of days and hijack the formatting whenever the ETA for banking and experimentation research are onscreen

**Comment by Tsangares** [comment]
Wait @AStonedApe isn't there already a way to change the time?

**Reply by Skilly146** [reply]
This would be news to me. There's nothing about it under options or on the research screen

**Reply by Tsangares** [reply]
Try the hotkey '%'

**Reply by Tsangares** [reply]
Does that do what you want?

**Reply by Tsangares** [reply]
There are maybe a bit too many features we left undocumented...

**Reply by AStonedApe** [reply]
'%' changes from 'x hours x minutes x seconds' to 'day @ time'. I think @Skilly146 is asking for a 3rd option, something like 'x cycles x ticks', perhaps.

**Reply by AStonedApe** [reply]
Maybe we could add to the '%' hotkey with a third option, and the hotkey could cycle through all of them.

**Reply by Tsangares** [reply]
Or an option in the settings????

**Reply by Skilly146** [reply]
The only SAT tools got keys I knew existed were the clipboard related ones transferred from NPAgent. I don't have my laptop with me but I'll try that tonight

**Reply by AStonedApe** [reply]
> Or an option in the settings????

Both?

**Reply by AStonedApe** [reply]
> The only SAT tools got keys I knew existed were the clipboard related ones transferred from NPAgent. I don't have my laptop with me but I'll try that tonight

'%' is from NPAgent. I have a list somewhere of all the hotkeys, there's a lot and many are kinda useless imo. Also, the ledger hotkey is 'm'.

**Comment by melonplant** [comment]
spent some time on this. with changes, pressing % will cycle through 3 time states for turn based or for real time.

state 0 - standard
state 1 - turn based: standard with turn(s) appended
![image](https://user-images.githubusercontent.com/100806306/208741140-73ef4c80-8e33-44cc-b1e0-c70c8ba2de73.png)
state 1 - real time: ETA time with clock (this was already in)
![image](https://user-images.githubusercontent.com/100806306/208741395-0479ee28-5541-4c46-a9c5-d442b61a3811.png)
state 2 - turn based: cycle time - standard with cycle(s) appended
![image](https://user-images.githubusercontent.com/100806306/208741666-d594d6f9-c5f4-4c00-8eae-b0d0c6fc207e.png)
state 2 - real time: cycle time - shows ticks/cycles to complete research. ETAs on ships are left as time
![image](https://user-images.githubusercontent.com/100806306/208745341-0b4c99a2-2741-41ad-9706-e229e9228039.png)
